### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by bsipocz**
Since the packages has been moved to use the astropy package template, `distribute_setup.py` is not being used. `cextern` is then an unused part of the template.

### CODE REVIEW

**Comment by cbanek** [comment]
I think this all looks good but now the build is red.  It looks like flake8 is picking up some style errors in the ci-helpers folder now that we're running it on the whole directory, and ci-helpers is checked out as a part of that build.  Although it always seems like a good idea to fix those, and they seem reasonable:

./ci-helpers/test_env.py:79:1: E302 expected 2 blank lines, found 1
./ci-helpers/test_env.py:93:37: W605 invalid escape sequence '\.'
./ci-helpers/test_env.py:93:45: W605 invalid escape sequence '\.'
./ci-helpers/test_env.py:123:13: F401 'pytest_doctestplus' imported but unused
./ci-helpers/test_env.py:191:13: F401 'sphinx' imported but unused
./ci-helpers/test_env.py:196:9: F401 'psutil' imported but unused
./ci-helpers/test_env.py:203:13: F401 'numpy' imported but unused
./ci-helpers/test_env.py:204:9: E722 do not use bare 'except'
./ci-helpers/test_env.py:232:13: F401 'scipy' imported but unused

It seems like it might be that way for a reason.  Maybe we should exclude the ci-helpers folder, as it isn't project code anyway?

**Comment by bsipocz** [comment]
> It seems like it might be that way for a reason. Maybe we should exclude the ci-helpers folder, as it isn't project code anyway?

Good point, `ci-helpers` is cloned into that directory. Also, I'm adding more stuff here just having some issues with figuring out the user-agent.

File: .travis.yml [hunk 1]
```diff
@@ -27,20 +27,19 @@ env:
         # The following versions are the 'default' for tests, unless
         # overridden underneath. They are defined here in order to save having
         # to repeat them for all configurations.
-        - PYTHON_VERSION=3.6
+        - PYTHON_VERSION=3.7
         - NUMPY_VERSION=stable
         - ASTROPY_VERSION=stable
         - MAIN_CMD='python setup.py'
         - SETUP_CMD='test'
-        - PIP_DEPENDENCIES='requests mimeparse requests_mock flake8'
```
Comment (cbanek) [inline .travis.yml]: Why get rid of flake8 here?  Is it installed by default in 3.7?

File: .travis.yml [hunk 2]
```diff
@@ -65,22 +64,19 @@ matrix:
         - os: linux
           env: ASTROPY_VERSION=development
                EVENT_TYPE='pull_request push cron'
-        - os: linux
-          env: PYTHON_VERSION=2.7 ASTROPY_VERSION=lts
-               PIP_DEPENDENCIES='requests mimeparse requests_mock contextlib2'
```
Comment (cbanek) [inline .travis.yml]: Is this just a copy of the same thing that's run on line 61 here?  I don't understand what the EVENT_TYPE means here in this case though.

File: pyvo/conftest.py [hunk 3]
```diff
@@ -1,10 +1,53 @@
...
+    del PYTEST_HEADER_MODULES['Scipy']
+    del PYTEST_HEADER_MODULES['Matplotlib']
+    del PYTEST_HEADER_MODULES['Pandas']
+except (NameError, KeyError):
     pass
+
+enable_deprecations_as_exceptions()
+
+
+
```
Comment (cbanek) [inline pyvo/conftest.py]: Okay this all seems complicated and I don't even pretend to understand it, but I'm guessing this is some kind of boilerplate.

File: pyvo/samp.py [hunk 4]
```diff
@@ -10,9 +10,9 @@
 import tempfile
 
 try:
-    from astropy.vo.samp import SAMPIntegratedClient
-except ImportError:
     from astropy.samp import SAMPIntegratedClient
+except ImportError:
+    from astropy.vo.samp import SAMPIntegratedClient
```
Comment (cbanek) [inline pyvo/samp.py]: Is this another one of those it was in one place in one version of astropy and somewhere different in another version?  If so, is this the time to use the minversion thing to make it a bit more explicit?

File: .travis.yml (hunk 1 above)
Comment (bsipocz) [reply .travis.yml]: ci-helpers installs it

File: .travis.yml (hunk 2 above)
Comment (bsipocz) [reply .travis.yml]: yes, this is a copy. the other one also runs from cron, it doesn't really have to though as python2 and dependencies is not moving anywhere anymore. The whole travis matrix could use a proper cleanup, adding stages etc, but that's beyond the scope of this PR.

File: pyvo/conftest.py (hunk 3 above)
Comment (bsipocz) [reply pyvo/conftest.py]: copy from the package template, these all adds nice testing headers, plugins, etc.

File: pyvo/samp.py (hunk 4 above)
Comment (bsipocz) [reply pyvo/samp.py]: that's possible sure, I simply kept the original logic but fixed the usage of it.

**Comment by cbanek** [comment]
So I'm still not sure why the build is broken, although I'm wondering if maybe it is related to the .travis.yml:30?  It looks like we're changing the default python version for the test --coverage run, which seems to be the only change that should affect things.  I'm not sure why that error is complaining about \w, which seems like a valid regexp, although maybe there is some other decorator we're supposed to put on that string?

As for the 2.7 failure, maybe just some dependency updated and broke us?  (Although then I'm not sure why we don't see this happen in other builds)

DeprecationWarning: The `implprefix` system is deprecated please decorate this function using an instance of HookimplMarker.

Seems like we could add that, but I'm not sure exactly where to add it to...

**Comment by cbanek** [comment]
This is great, thanks!

**Pull request merged**